README.Rmd

```diff
@@ -0,0 +1,191 @@
+scripts". This is misleading because this workflow can run offline too.
+* __Multiple-outputs workflow__: When the number of portfolios and/or investors
+is not exactly one, we use a different set of optimized scripts. This workflow
+is typically used to produce static (.pdf) reports. By contrast with the
+so-called "online" workflow, this one has been usually referred to as "the
+offline workflow"; again, this is misleading.
+
+Both workflows include three steps:
+
+1. Clean portfolio: Clean the input portfolio and merge in financial data, to
+categorize each holding and identify whether it's equity (EQ) or corporate bonds
+(CB).
+2. Run Analysis: Merge portfolios with asset-level data and scenarios, then
+group results at company, portfolio and regional level.
+3. Present results: Present the results in a clear output format.
+
+--
+
+Note: Code chunks with prompts "`$`" and "`#>`" correspond to bash and R,
+respectively.
+
+### Siblings
+
+PACTA_analysis depends on other "siblings" of the PACTA family. They are all
+available at <https://github.com/2DegreesInvesting> and should live alongside
+PACTA_analysis/, under the same parent directory. All siblings include an empty file called ".pacta", which you can use to find them programmatically.
+
+```{bash comment="$"}
+siblings="$(dirname $(find .. -maxdepth 2 -empty -type f -name '.pacta'))"
+echo $siblings
+```
+
+Before any analysis, ensure all siblings are up to date.
+
+```{bash comment="$"}
+# Store
+ working_directory="$pwd"
+
+siblings="$(dirname $(find .. -maxdepth 2 -empty -type f -name '.pacta'))"
+for i in $siblings; do
+  cd $i && echo $i
+  git log --decorate --oneline -n 1
+done
+
+# Restore
+cd "$working_directory"
+
+```
+
+For an easier way to work with all siblings at once from the terminal see
+[pacta-cli](https://github.com/2DegreesInvesting/pacta-cli).
+
+## Computing environment
+
+The required computing environment is available in [this docker
+image](https://github.com/2DegreesInvesting/docker/tree/master/r-packages).
+
+Other computing environments may be defined as workflows for GitHub actions.
+
+```{bash comment="$"}
+ls .github/workflows
+```
+
+Unless otherwise indicated, R packages should be up to date.
+
+<details> <summary>Session information.</summary>
+```{r}
+devtools::session_info()
+```
+</details>
+
+As an example, this is one way to use that computing environment to produce the
+very document you are now reading:
+
+* I move to the parent directory of PACTA_analysis and its siblings.
+
+```bash
+cd ~/git
+```
+
+* I run (`docker run`) an ephemeral (`--rm`), interactive (`-ti`) container of
+the image `2dii/r-packages`; create a number of volumes (`-v`) to access useful
+files in my host computer from inside the container; and set the working
+directory (`-w`) to PACTA_analysis/.
+
+```{r comment="", echo=FALSE}
+url <- "https://raw.githubusercontent.com/maurolepore/bin/master/docker-run-2dii-r-packages"
```

Comment:
I would rather see this code spelled out. Loading a script from a web URL and blindly running it sends chills down my spine

Reply:
I agree. This code chunk uses `echo=FALSE`. The [rendered file](https://github.com/2DegreesInvesting/PACTA_analysis/blob/generalize-workflow/README.md) looks like this:

![image](https://user-images.githubusercontent.com/5856545/109856455-5c9b4c80-7c0e-11eb-8426-1bb686d2feff.png)

Reply:
ok, but.... if a user renders this Rmd file (which is the intent, or not?), then their computer needs to pull this script from a remote location that's not even on at a 2dii owned URL, right? The script is hosted on a personal repo of yours. Could it be moved to a 2dii repo so that it doesn't look suspicious? Or why not just copy paste that code directly into this README here? Do we really want this README to be able to be changed by an external repo?

I'm trying to imagine being an external user reading this (this repo is public) and how I would interpret what's happening here or what concerns I would have. I guess you're expecting that a user would usually only be reading the pre-rendered markdown version of this so it doesn't matter? But if a change is ever made to this README, then whoever does that will have to re-render it (or maybe a GH action will rerender it) and at that point this remote, external repo will be a hard requirement, right?